src/ripple/rpc/ShardVerificationScheduler.h

```diff
@@ -0,0 +1,84 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_RPC_SHARDVERIFICATIONSCHEDULER_H_INCLUDED
+#define RIPPLE_RPC_SHARDVERIFICATIONSCHEDULER_H_INCLUDED
+
+#include <ripple/app/main/Application.h>
+#include <functional>
+
+namespace ripple {
+namespace RPC {
+
+class ShardVerificationScheduler
+{
+public:
+    // This is the signature of the function that the client
+    // wants to have invoked upon timer expiration. The function
+    // should check the error code 'ec' and abort the function
+    // if the timer was cancelled:
+    // (ec == boost::asio::error::operation_aborted).
+    // In the body of the function, the client should perform
+    // the necessary verification.
+    using retryFunction =
+        std::function<void(boost::system::error_code const& ec)>;
+
+    ShardVerificationScheduler() = default;
```

Comment:
The default constructed one is viable so I see no reason to delete it. In fact deleting it, might be confusing, making people wonder why its deleted at all. Mitigating this might require a comment, adding information and complexity that could just be avoided.

Reply:
It is viable but un-needed and potentially harmful.  Default constructors can be invoked in all sorts of circumstances.  It seems to me that, given the current design, you always want the two arguments passed in to the `ShardVerificationScheduler` by whoever is constructing it.  Yes, you have defined reasonable defaults but they are currently neither used nor tested.

Another safe choice would be to simply delete the line declaring the default constructor.  Since you're declaring a (different) constructor that takes arguments the compiler will not generate the default constructor for you unless someone asks (as you have here).  Deleting the line would require no additional comments or explanations.

Reply:
The default values are fine. They are currently used and tested since I don't provide configuration values during verification testing. This means the values passed to the constructor are 0, which causes the the defaults to be used.

Reply:
:shrug: I'm not gonna fight you about it.  I just think leaving around unnecessary constructors is asking for trouble.  Your call.